docs.go

```diff
@@ -1,31 +1,62 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: MPL-2.0
 
-/*
-Package mql provides a language that end users can use to query your
-database models, without them having to learn SQL or exposing your
-application to SQL injection.
-
-You define a model you wish to query as a Go struct and provide a mql query. The
-package then uses the query along with a model to generate a parameterized SQL
-where clause.
-
-Fields in your model can be compared with the following operators:
-=, !=, >=, <=, <, >, %
-
-Double quotes `"` can be used to quote strings.
-
-Comparison operators can have optional leading/trailing whitespace.
-
-The % operator allows you to do partial string matching using LIKE and this
-matching is case insensitive.
-
-The = equality operator is case insensitive when used with string fields.
-
-Comparisons can be combined using: and, or.
-
-More complex queries can be created using parentheses.
-
-Example: name=alice and age > 11 and (region % Boston or region="south shore")
-*/
+// Package mql provides a language that end users can use to query your
+// database models, without them having to learn SQL or exposing your
+// application to SQL injection.
+//
+// # Examples
+//
+// Gorm: https://github.com/go-gorm/gorm
+//
+//	w, err := mql.Parse("name=alice or name=bob)",User{})
+//	  if err != nil {
+//	  return nil, err
+//	}
+//	err = db.Where(w.Condition, w.Args...).Find(&users).Error
```

Comment:
Code blocks in Go code should just use a single space indentation: https://go.dev/doc/comment#code.
```suggestion
//  w, err := mql.Parse("name=alice or name=bob)",User{})
//  if err != nil {
//  	return nil, err
//  }
//  err = db.Where(w.Condition, w.Args...).Find(&users).Error
```

I think gofmt should format these automatically though 🤔

Reply:
gofmt did format them when I saved my editor.

Reply:
If nothing else, the alignment on the `if` in this block is off 😁

Reply:
Sorry, it should be indented by a single tab only, apparently:
> Gofmt indents all lines in a code block by a single tab

Reply:
In that case I think this should just adjust the if statement indentation

Reply:
You might want to replace the two spaces in the returns with a tab as well but that's a nit

Reply:
This if statement is still misaligned